rules/order/src/StmtVisibilitySorter.php

```diff
@@ -86,42 +72,24 @@ public function sortMethods(ClassLike $classLike): array
             /** @var string $classMethodName */
             $classMethodName = $this->nodeNameResolver->getName($classStmt);
 
-            $classMethods[$classMethodName][self::NAME] = $classMethodName;
-            $classMethods[$classMethodName][self::VISIBILITY] = $this->getVisibilityLevelOrder($classStmt);
-            $classMethods[$classMethodName]['abstract'] = $classStmt->isAbstract();
-            $classMethods[$classMethodName]['final'] = $classStmt->isFinal();
-            $classMethods[$classMethodName]['static'] = $classStmt->isStatic();
-            $classMethods[$classMethodName][self::POSITION] = $position;
+            $classMethodsRankeables[] = new ClassMethodRankeable(
+                $classMethodName,
+                $this->getVisibilityLevelOrder($classStmt),
+                $position,
+                $classStmt
+            );
         }
 
-        uasort(
-            $classMethods,
-            function (array $firstArray, array $secondArray): int {
-                return [
-                    $firstArray[self::VISIBILITY],
-                    $firstArray['static'],
-                    $secondArray['abstract'],
-                    $firstArray['final'],
-                    $firstArray[self::POSITION],
-                ] <=> [
-                    $secondArray[self::VISIBILITY],
-                    $secondArray['static'],
-                    $firstArray['abstract'],
-                    $secondArray['final'],
-                    $secondArray[self::POSITION],
-                ];
-            }
-        );
-
-        return $classMethods;
+        return $this->sortByRanksAndGetNames($classMethodsRankeables);
```

Comment:
@dobryy One last trick that I just learn to use (PHPStan rules in progress ;)): value objects beat arrays.

Here all the sorting arrays with ranks could be refactored to value objects.

I can see the common piece to all rules is "rank" (any rank).

So `RankeableInterface` can be used to sort them all and `sortByRanksAndGetNames()` can handle all the sorting stuff.